src/box/lua/schema.lua

```diff
@@ -2046,8 +2049,14 @@ local function check_pairs_opts(opts, key_is_nil, level)
                 box.error(box.error.ITERATOR_POSITION, level and level + 1)
             end
         end
+        if opts.offset ~= nil then
+            offset = opts.offset
+            if type(offset) ~= "number" then
+                box.error(box.error.INVALID_OFFSET, level and level + 1)
+            end
+        end
```

Comment:
We don't have a separate error code for this in `select`. I don't see a point in adding one for `pairs`. `INVALID_ARG` should do.

Reply:
Done, the error is catched on the Lua side now (as it is for the regular select).